src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -961,25 +961,79 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
     return matched
 
 
-def create_function(resource_group_name, new_app_name, storage_account, plan=None):
+def create_function(resource_group_name, name, storage_account, plan=None,
+                    consumption_plan_location=None):
+
+    if plan is not None and consumption_plan_location is not None:
+        raise CLIError("Use Plan or Consumption Plan Location")
```

Comment:
Raising a CLIError, you'll get the same error in a validator.  If it were a ValueError it was display the usage string, but really, that's not going to reflect this logic anyways. I would suggest making the error look more like a usage string though.
`usage error: --plan NAME_OR_ID | --consumption-plan-location LOCATION`